webapp-django/crashstats/crashstats/models.py

```diff
@@ -609,93 +630,72 @@ class TelemetryCrash(SocorroMiddleware):
     aliases = {"crash_id": "uuid"}
 
 
+class PermissionsReducerError(Exception):
+    pass
+
+
+@functools.lru_cache
+def get_processed_crash_permissions_reducer(permissions_tuple):
```

Comment:
The variations of permissions_tuple values is pretty minimal, so this will do some work up front and then keep that cached for the duration of the process.